### ISSUE TITLE
Cannot a #[pymodule] function as doc(hidden) since 0.14

### ISSUE THREAD

**Issue by adamreichold** [PyO3/pyo3#1720]
We basically use the following entry point for a Python extension
```rust
use pyo3::prelude::*;

#[doc(hidden)]
#[pymodule]
#[pyo3(name = "swifco_rs")]
fn init(_py: Python, _m: &PyModule) -> PyResult<()> {
    Ok(())
}
```
but this fails to compile after upgrading to version 0.14 with the following error
```
error: invalid doc comment
 --> src/lib.rs:3:1
  |
3 | #[doc(hidden)]
  | ^^^^^^^^^^^^^^
```

(We use `--document-private-items` as we use the doc comments to explain the implementations details of this project and hence like to explicitly hide any boiler plate like these module initialisation functions.)

**Comment by davidhewitt** [comment]
Thanks for the report! I agree this is a regression, and have opened #1722 to fix.

Appreciate this is an upgrade blocker for you, however I'll give this a bit of time to sit on `main` before releasing so that other fixes (if needed) have time to collect.

**Comment by adamreichold** [comment]
> Appreciate this is an upgrade blocker for you, however I'll give this a bit of time to sit on main before releasing so that other fixes (if needed) have time to collect.

Thank you for the quick response and resolution! I did want to convey a sense of urgency and I am sorry if I did. (Upgrading our simulation is actually blocked on a new release of `rust-numpy` on crates.io that is compatible with version 0.14. But then again, we face no functional issues with version 0.13 and just try to keep up with the ecosystem.)
